container/kvm/wrappedcmds.go

```diff
@@ -147,69 +148,49 @@ func CreateMachine(params CreateMachineParams) error {
 	if params.Hostname == "" {
 		return fmt.Errorf("hostname is required")
 	}
-	if params.pathfinder == nil {
-		params.pathfinder = paths.DataDir
-	}
 
-	// We need to run some commands as root and some as libvirt. Alternatively,
-	// we could just run commands as root then change ownership as nessecary,
-	// but that's just gross in a different way.
-	var (
-		runCmdAsRoot runFunc
-	)
-	// if there isn't a runCmd passed in, we're not testing so use real
-	// runFuncs.
+	if params.findPath == nil {
+		params.findPath = paths.DataDir
+	}
 	if params.runCmd == nil {
-		runCmdAsRoot = run
 		params.runCmd = runAsLibvirt
-	} else {
-		// If there is a runCmd passed in we're passing a test stub. So set
-		// runCmdAsRoot to use it too.
-		runCmdAsRoot = params.runCmd
+	}
+	if params.runCmdAsRoot == nil {
+		params.runCmdAsRoot = run
```

Comment:
Thanks - I think I was just confused because of the way the defaulting was intertwined before.